scripts/maint/release.py

```diff
@@ -132,12 +132,36 @@ def main(args):
     # While we use Github releases and not dist.tpo
     print("\n6. Create release tarball")
     print("-------------------------")
+
+    print("\nCreating a release tarball...")
+    subprocess.call(
+        "git archive --format=tar.gz --prefix=sbws-{}/ "
+        "-o v{}.tar.gz v{}"
+        .format(release_version, release_version, release_version).split(' ')
+    )
+    print("\nCreating tarball hash file...")
+    fd = open('v{}.tar.gz.sha256'.format(release_version), 'w')
+    subprocess.call("sha256sum v{}.tar.gz".format(release_version).split(' '),
+                    stdout=fd)
+    fd.close()
+
     print("Obtaining Github tarball...")
+    # This will overwrite local tarball, but that's fine since the hash file
+    # won't be overwritten.
     subprocess.call(
-        "torsocks wget https://github.com/torproject/sbws/archive/v{}.tar.gz"
-        .format(release_version).split(' ')
+        "wget https://github.com/torproject/sbws/archive/v{}.tar.gz "
+        "-O v{}.tar.gz"
+        .format(release_version, release_version).split(' ')
         )
 
+    print("Verifying Github tarball and local one are the same...")
```

Comment:
I don't understand why we do a sha256 check, rather than a file diff check.
In fact, I'm not sure why we do this check at all?
Please add a comment explaining why it is important that GitHub matches our local tarball.

Are tarballs reproducible?
What happens if the tarballs are different?
Can they be different on different OSes?
Does GitHub guarantee that their tarballs are created with particular git and tar versions on a particular OS?